packages/api/resolvers/mutations/addProductAssignment.js

```diff
@@ -11,11 +16,17 @@ export default function addProductAssignment(
 
   if (!proxyId) throw new InvalidIdError({ proxyId });
   if (!productId) throw new InvalidIdError({ productId });
+  const product = Products.findOne({ _id: productId });
+  if (!product) throw new ProductNotFoundError({ productId });
+  if (product.type !== ProductTypes.ConfigurableProduct)
```

Comment:
I think this is wrong, proxyId product should be asserted as a configurable product, maybe use childProduct as variable name instead of product to not have us fall into this again.